### PR TITLE
Feature/daemon modal

### PR DESCRIPTION
**Pull request by pciavald**

### CODE REVIEW

File: src/app/core/rpc/rpc.module.ts [hunk 1]
```diff
@@ -21,8 +21,7 @@ export class RpcModule {
         RPCService,
         PeerService,
         PassphraseService,
-        BlockStatusService,
-        ElectronService
```
Comment (pciavald) [inline src/app/core/rpc/rpc.module.ts]: appeared twice

File: src/app/core/rpc/rpc.service.ts [hunk 2]
```diff
@@ -127,8 +154,14 @@ export class RPCService {
  * @returns      void
  */
 
-  register(instance: Injectable, method: string, params: Array<any> | Function | null,
-           successCB: Function, when: string, errorCB?: Function): void {
```
Comment (pciavald) [inline src/app/core/rpc/rpc.service.ts]: readability

File: src/app/core/rpc/rpc.service.ts [hunk 3]
```diff
@@ -68,7 +79,13 @@ export class RPCService {
  * @returns      void
  */
 
-  call(instance: Injectable, method: string, params: Array<any> | null, successCB: Function, errorCB?: Function): void {
```
Comment (pciavald) [inline src/app/core/rpc/rpc.service.ts]: readability

File: src/app/core/rpc/rpc.service.ts [hunk 4]
```diff
@@ -88,13 +105,23 @@ export class RPCService {
...
           response => {
             successCB.call(instance, response.json().result);
+            this.modalUpdates.next({
+              response: response,
+              electron: this.isElectron
+            });
           },
           error => {
             if (errorCB) {
-              errorCB.call(instance, (typeof error['_body'] === 'object' ? error['_body'] : JSON.parse(error['_body'])) )
```
Comment (pciavald) [inline src/app/core/rpc/rpc.service.ts]: readability

File: src/app/modals/modals.service.ts [hunk 5]
```diff
@@ -71,7 +82,9 @@ export class ModalsService {
 
   needToOpenModal(status: any) {
     // Open syncing Modal
-    if (!this.isOpen && (status.networkBH <= 0 || status.internalBH <= 0 || status.networkBH - status.internalBH > 50)) {
```
Comment (pciavald) [inline src/app/modals/modals.service.ts]: readability

**Comment by pciavald** [comment]
travis fails because of a bug in angular-cli : https://github.com/angular/angular-cli/issues/7296

**Comment by rynomster** [comment]
Could we get the lint fixed? :D

**Comment by pciavald** [comment]
Oopsie ! Strange my linter didn't complain.  jeudi, 10 août 2017, 09:33AM +02:00 de Ryno  notifications@github.com :

File: src/app/modals/daemon/daemon.component.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  selector: 'app-daemon',
+  templateUrl: './daemon.component.html',
+  styleUrls: ['./daemon.component.scss']
+})
+export class DaemonComponent {
+
+  public message: any;
+
+  constructor(
+    @Inject(forwardRef(() => ModalsService)) private _modalsService: ModalsService
```
Comment (kewde) [inline src/app/modals/daemon/daemon.component.ts]: Seems unused? triggering moved to the model.service

**Pull request merged**